src/tools/qt-gui/src/undomanager.cpp

```diff
@@ -47,10 +47,16 @@ void UndoManager::createEditKeyCommand(TreeViewModel* model, int idx, QVariantLi
 	TreeViewModel* tmpModel = qvariant_cast<TreeViewModel*>(data.takeAt(2));
 	QVariantMap oldMDMap;
 
-	foreach (ConfigNodePtr node, tmpModel->model())
+	if(tmpModel)
 	{
-		oldMDMap.insert(node->getName(), node->getValue());
+
+		foreach (ConfigNodePtr node, tmpModel->model())
+		{
+			oldMDMap.insert(node->getName(), node->getValue());
+		}
 	}
+	else
+		tmpModel = new TreeViewModel;
 
 	data.insert(2, oldMDMap);
```

Comment:
0003088 wrote:
> You complained that the signature of the command was too long, so I put the
> parameters into a map. Of course the index needs stay the same if a
> parameter is edited. I will add a comment that maps the indices to the
> items (or return to the long signature?).

I am not sure if map is a good idea. A class/struct is more appropriate to 
avoid too many parameters.

The avoidance of magic values is, however, independent of this issue.